From b9b4facc9d6c849bb0e2ba27cb5a5f48f28c07ab Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 14 Mar 2022 06:13:44 +0200
Subject: [PATCH] [SW-56706] rom: spi internal dma api fix

A dedicated bit in the dma ctrl register needs to be set based
on destination type is SRAM. This patch fixes the same in the dma
internal function.

Change-Id: I71714f27bc291c664ef00fad4f03b92ad171e000
---

diff --git a/arch/gaudi2/arc/regmask.h b/arch/gaudi2/arc/regmask.h
index 23df57c..4a1e217 100644
--- a/arch/gaudi2/arc/regmask.h
+++ b/arch/gaudi2/arc/regmask.h
@@ -7,6 +7,7 @@
 #define _GAUDI2_REGMASK_H_
 
 #define SPI_DMA_CTRL_ADDR_SHIFT			PSOC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT
+#define SPI_DMA_CTRL_DST_SRAM_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_CTRL_DST_SRAM_SHIFT
 #define SPI_DMA_STATUS_ERROR_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_MASK
 #define SPI_DMA_STATUS_ERROR_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_SHIFT
 #define SPI_DMA_STATUS_DONE_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_MASK
diff --git a/arch/gaudi3/arc/regmask.h b/arch/gaudi3/arc/regmask.h
index ee7aa50..0bc5e6a 100644
--- a/arch/gaudi3/arc/regmask.h
+++ b/arch/gaudi3/arc/regmask.h
@@ -7,6 +7,7 @@
 #define _GAUDI3_REGMASK_H_
 
 #define SPI_DMA_CTRL_ADDR_SHIFT			D0_PARC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT_S
+#define SPI_DMA_CTRL_DST_SRAM_SHIFT		D0_PARC_GLOBAL_CONF_SPI_DMA_CTRL_DST_SRAM_S
 #define SPI_DMA_STATUS_ERROR_MASK		D0_PARC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_M
 #define SPI_DMA_STATUS_ERROR_SHIFT		D0_PARC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_S
 #define SPI_DMA_STATUS_DONE_MASK		D0_PARC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_M
diff --git a/arch/greco/arc_elf32/regmask.h b/arch/greco/arc_elf32/regmask.h
index 9428755..1e9c392 100644
--- a/arch/greco/arc_elf32/regmask.h
+++ b/arch/greco/arc_elf32/regmask.h
@@ -7,6 +7,7 @@
 #define _GRECO_REGMASK_H_
 
 #define SPI_DMA_CTRL_ADDR_SHIFT			PSOC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT
+#define SPI_DMA_CTRL_DST_SRAM_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_CTRL_DST_SRAM_SHIFT
 #define SPI_DMA_STATUS_ERROR_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_MASK
 #define SPI_DMA_STATUS_ERROR_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_SHIFT
 #define SPI_DMA_STATUS_DONE_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_MASK
diff --git a/common/dma.c b/common/dma.c
index 98aad95..db690f2 100644
--- a/common/dma.c
+++ b/common/dma.c
@@ -26,8 +26,8 @@
 	/* enable DMA transfer */
 	reg_spi_dma_ctrl_s = SPI_DMA_ENABLE;
 
-	/* fw_sram as destination */
-	reg_spi_dma_ctrl_s |= fw_sram_dst;
+	/* sram=1 hbw=0 as destination */
+	reg_spi_dma_ctrl_s |= fw_sram_dst << SPI_DMA_CTRL_DST_SRAM_SHIFT;
 
 	/* # of 4bytes to DMA copy */
 	reg_spi_dma_ctrl_s |= (uint32_t)image_size_4b << fld_size_shift;
